react/features/app/actions.native.ts

```diff
@@ -168,6 +169,12 @@ export function appNavigate(uri?: string, options: IReloadNowOptions = {}) {
                 navigateRoot(screen.unsafeRoomWarning);
             } else {
                 navigateRoot(screen.preJoin);
+
+                // On mobile the {@link isPrejoinPageVisible} will always return false. This is the first place we
+                // currently know that the prejoin has been displayed.
+                dispatch(setPermanentProperty({
```

Comment:
You need to move this to above the if. The prejoin screen will also be displayed right after the unsafe room warning screen.

Alternatively, you could do it in the actual prejoin screen.

Reply:
Nice catch!

I like the idea for setting it trough the prejoin screen more.